esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -0,0 +1,438 @@
+"""Implementation of the climwip weighting scheme.
+
+Lukas Brunner et al. section 2.4
+https://iopscience.iop.org/article/10.1088/1748-9326/ab492f
+"""
+from collections import defaultdict
+import os
+import logging
+from datetime import datetime
+
+import yaml
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+import xarray as xr
+from scipy.spatial.distance import pdist, squareform
+
+from esmvaltool.diag_scripts.shared import (ProvenanceLogger,
+                                            get_diagnostic_filename,
+                                            get_plot_filename, run_diagnostic,
+                                            select_metadata)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def get_provenance_record(caption: str, ancestors: list):
+    """Create a provenance record describing the diagnostic data and plots."""
+    record = {
+        'caption': caption,
+        'domains': ['reg'],
+        'authors': [
+            'kalverla_peter',
+            'smeets_stef',
+            'brunner_lukas',
+            'camphuijsen_jaro',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': ancestors,
+    }
+    return record
+
+
+def log_provenance(caption: str,
+                   filename: str,
+                   cfg: dict,
+                   provenance_info: list):
+    """Log provenance info."""
+    provenance_record = get_provenance_record(caption,
+                                              ancestors=provenance_info)
+    with ProvenanceLogger(cfg) as provenance_logger:
+        provenance_logger.log(filename, provenance_record)
+
+
+def read_metadata(cfg: dict, projects: list) -> dict:
+    """Read the metadata from the config file.
+
+    Project is the list of project identifiers (str) to select.
+    """
+    datasets = defaultdict(list)
+
+    for project in projects:
+        metadata = select_metadata(cfg['input_data'].values(), project=project)
+
+        for item in metadata:
+            variable = item['short_name']
+
+            datasets[variable].append(item)
```

Comment:
I think you could probably use `esmvaltool.diag_scripts.shared.group_metadata` to do this too, but it's not that important

Reply:
Yeah, it does something similar, but it is not exactly the same. So we didn't change anything.